### DISCUSSION TITLE
Plese tell me about OMR user id.

### DISCUSSION THREAD

**Discussion by sasta-2024** [sassoftware/saspy#613] [Q&A]
When I try to access the sas server(remote) using saspy, I am asked to enter my OMR user id.

>>> import saspy
>>> sas = saspy.SASsession(cfgname='winiomwin')
Please enter the OMR user id:

I'm a newbie and need some details.

**Comment by tomweber-sas** [comment]
Hey @sasta-2024 sure. To connect to this kind of SAS deployment, you need credentials. These are Metadata Server Credentials. If you have ever connected to this system with any other client, you provided these in some way. This may be overkill, but here's a document that talks about it: https://go.documentation.sas.com/doc/en/bicdc/9.4/bisecag/n1g71bizsmlh47n199hkr4edkg8j.htm
SASPy goes about this a little different that what's explained in there, but it's all the same thing. I don't interact with the Metadata Server directly. There's another server in this kind of deployment which is called the Object Spawner, it connects clients to various other servers (Workspace servers, Stored Porcess servers, ...). Even clients that do interact with the metadata server, still connect to the Object Spawner to get access to the server they want to interact with. So the OMR (Open Metadata Repository; aka Metadata Server) userid and password are you're credentials to the SAS System, same no matter what client you're trying to connect with.

**Reply by sasta-2024** [reply]
Thanks, Tom.
I greatly appreciate your thoughtful reply.
I understood something about the OMR (Open Metadata Repository; aka Metadata Server) userid and password are you're credentials to the SAS System.
When I enter my userid, it does not respond, but I cannot determine what is going wrong.
I would appreciate it if you could provide me with any information needed.

**Reply by tomweber-sas** [reply]
Well, I don't see what you're doing, so I don't know what's going on. Do you not get prompted for the pw after you provide your id? That would be next if you haven't provided that via one of the various way you could (since you're being prompted for the id, I have to assume you haven't). What is your configuration definition you're using? Have you ever connect to this SAS system with any other client? EG perhaps? 
Can you provide any information about what you're trying so I can can help?

**Reply by sasta-2024** [reply]
Thanks, Tom.
What I would like to do is use saspy to run SAS on a remote desktop.
After entering the id, it does not prompt me for the pw.
I have never connected to this SAS system with any other client.
Can I just put sascfg_personal.py for the configuration definition?

I would appreciate a detailed explanation as I am still learning.

**Comment by tomweber-sas** [comment]
I still don't have a good understanding of your situation. What kind of SAS deployment are you trying to connect to? Do you have credentials? Do you have the configuration information you need to be able to connect? Have you read the [configuration instructions](https://sassoftware.github.io/saspy/configuration.html)? Which scenario [here](https://sassoftware.github.io/saspy/configuration.html#choosing-an-access-method) are you trying to implement?
What is your configuration you're trying to use currently? You show `sas = saspy.SASsession(cfgname='winiomwin')`, but what is your `winiomwin` configuration? That's just a variable name, what do you actually have defined for that?
Do you have an SAS Admin that set you up to be able to connect to this deployment? Is it a deployment, or just a stand alone PC/SAS install?
I need more information of your environment/situation to have any real understanding of how you need to connect to this SAS system. As you can see in the documentation, there are multiple different ways depending upon you SAS deployment and where you're running Python. The configuration doc explains all of these variations and which Access Methods can be used for which combinations.
